### ISSUE TITLE
[bug] When toggling true/false autoHeaderHeight and wrapHeaderText ColDef properties, the header styles get messed up. 

### ISSUE THREAD

**Issue by tielushko** [ag-grid/ag-grid#7155]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => see 'Providing a Reproducible Scenario'
[] feature request => do not use Github for feature requests, see 'Customers of AG Grid'
[] support request => see 'Requesting Community Support'
```
Current behaviour: 

I would like to report a bug with the autoHeaderHeight and wrapHeaderText properties. When toggling those two properties, after being false and becoming true, the styles for the header height get messed up. The initial header wrap and height = true works without any problems. Once the value is toggled to false, then back to true, the box gets messed up. 

Please see example for the bug in this codesandbox. To repro, click toggle to turn off the auto height and wrap, and then click again to turn it back on. Look at the Gold... column.

https://codesandbox.io/s/header-auto-height-toggle-bug-js57z3?file=/src/index.tsx

Expected behaviour: 
I think toggling the state for header height and wrap text would automatically show the text within the container of the header cell. 

**Please tell us about your environment:**
Please see a reproducible example in code sandbox.

* **AG Grid version:** 30.2.0 (latest)

* **Browser:**

using code sanbox inside Chrome. reproducible in all browsers.

* **Language:** 
typescript

**Comment by AG-Zoheil** [comment]
Hello 

Thank you for reporting this - much appreciated!

You'll be pleased to know that we have already discovered and fixed this bug. It will be part of our next release.

You can track it with the following reference: **AG-9461**

The best way to track this is to sign up for AG Grid new release notifications using [the instructions here](https://ag-grid.zendesk.com/hc/en-us/articles/360002648058-How-do-I-sign-up-for-ag-Grid-release-notifications-). This way you'll know as soon as a new version is out and you can check whether this specific item was implemented on [the changelog page](https://www.ag-grid.com/ag-grid-changelog/).

Thanks again for bringing this up with us.

Kind regards,
Zoheil

**Comment by AG-Zoheil** [comment]
Hello,

Thank you for reporting this - much appreciated!

Based on your description we've now reproduced this issue as a bug.

We have added this bug to our backlog and we are tracking it with the following reference: **AG-9595**

We try to fix bugs from one release to the next, so this should be fixed in the next release or the one after if it was raised too close to the next release date.

See whether this item will be in the next release by checking the NEXT RELEASE checkbox on the product pipeline page:
https://www.ag-grid.com/ag-grid-pipeline/

The best way to track this is to sign up for AG Grid new release notifications using [the instructions here](https://ag-grid.zendesk.com/hc/en-us/articles/360002648058-How-do-I-sign-up-for-ag-Grid-release-notifications-). This way you'll know as soon as a new version is out and you can check whether this specific item was implemented on [the changelog page](https://www.ag-grid.com/ag-grid-changelog/).

Thanks again for bringing this up with us.

Kind regards,
Zoheil

**Comment by tielushko** [comment]
@AG-Zoheil, thank you for your quick response. Would it be of any help to have the ticket prioritized through enterprise support zendesk submission?

**Comment by AG-Zoheil** [comment]
Hi @tielushko,

We generally encourage our Enterprise clients to contact us via our [support portal](https://ag-grid.zendesk.com/) for issues or suggestions. That would be the quickest and most efficient route for you to get your question answered.

Regarding the issue above, as it's already been logged as a bug now, it will be picked up in due time. We don't like to leave bugs hanging around for too long (there will be exceptions) and hence there is nothing further required from you.

Kind regards,
Zoheil

**Comment by tielushko** [comment]
@AG-Zoheil I was wondering if something happened to the AG-9595 since I no longer able to see it in the pipeline?

**Comment by AG-Zoheil** [comment]
Hi @tielushko,

AG-9595 was released with our 31.0.0 version at the end of November 2023. Please visit our [changelog](https://www.ag-grid.com/changelog/?fixVersion=31.0.0) or [blog](https://blog.ag-grid.com/whats-new-in-ag-grid-31/) to see the many improvements and fixes that we made.

Thanks!
